### PR TITLE
Checking whether launchEngineOpHandle is null for KyuubiConnection:: getExecLog

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

I am developing internal submit application operation and there is no backend engine needed to launch.
But I meet below exception.
```
00:07:40.744 KyuubiThriftBinaryFrontendServiceHandler-Pool: Thread-37 ERROR server.KyuubiThriftBinaryFrontendService: Error fetching results:
java.lang.NullPointerException
	at org.apache.kyuubi.service.ThriftBinaryFrontendService.FetchResults(ThriftBinaryFrontendService.scala:494)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$FetchResults.getResult(TCLIService.java:1837)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$FetchResults.getResult(TCLIService.java:1822)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:39)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
	at org.apache.kyuubi.service.authentication.TSetIpAddressProcessor.process(TSetIpAddressProcessor.scala:36)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:286)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
it was invoked in KyuubiCommands::showRemainingLogsIfAny

```
  private void showRemainingLogsIfAny(Object sqlObject) {
    if (sqlObject instanceof KyuubiLoggable) {
      KyuubiLoggable kyuubiLoggable = (KyuubiLoggable) sqlObject;
      List<String> logs = null;
      do {
        try {
          logs = kyuubiLoggable.getExecLog();
```

And if using the kyuubi beeline to connect old kyuubi server, it will meet the same issue.
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java [hunk 1]
```diff
@@ -237,6 +237,11 @@ public boolean hasMoreLogs() {
       throw new ClosedOrCancelledException(
           "Method getEngineLog() failed. The " + "connection has been closed.");
     }
+
+    if (launchEngineOpHandle == null) {
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:241]: Is this handle that caller call `getExecLog()` without check of `hasMoreLogs()`?

Comment (turboFei) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:241]: I see,

it was invoked in
```
  private void showRemainingLogsIfAny(Object sqlObject) {
    if (sqlObject instanceof KyuubiLoggable) {
      KyuubiLoggable kyuubiLoggable = (KyuubiLoggable) sqlObject;
      List<String> logs = null;
      do {
        try {
          logs = kyuubiLoggable.getExecLog();
```

Comment (turboFei) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:241]: here the method name is `showRemainingLogsIfAny`.
and hasMoreLogs will check whether the launch engine operation completed.

So, it does not check hasMoreLogs.

Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:241]: Is there any side effect if we check `hasMoreLogs` before call `showRemainingLogsIfAny`?

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java [hunk 2]
```diff
@@ -237,6 +237,11 @@ public boolean hasMoreLogs() {
       throw new ClosedOrCancelledException(
           "Method getEngineLog() failed. The " + "connection has been closed.");
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:238]: The method name is outdated

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java (hunk 1 above)
Comment (turboFei) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:241]: if we check hasMoreLogs here, the `showRemainingLogsIfAny` will not show remaining logs.

**Comment by turboFei** [comment]
with out this pr,
if the use this beeline to connect old kyuubi server will meet the same issue as described in pr description.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
CI failed with unrelated issue, thanks, merging to master!

**Pull request closed without merging**